src/datatypes/image_mode.rs

```diff
@@ -0,0 +1,127 @@
+use std::fmt::{Display, Formatter, Result};
+use std::str::FromStr;
+use std::string::ToString;
+
+#[cfg(feature = "python")]
+use pyo3::{exceptions::PyValueError, prelude::*};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    datatypes::DataType,
+    error::{DaftError, DaftResult},
+};
+
+/// Supported image modes for Daft's image type.
+///
+/// L       - 8-bit grayscale
+/// LA      - 8-bit grayscale + alpha
+/// RGB     - 8-bit RGB
+/// RGBA    - 8-bit RGB + alpha
+/// L16     - 16-bit grayscale
+/// LA16    - 16-bit grayscale + alpha
+/// RGB16   - 16-bit RGB
+/// RGBA16  - 16-bit RGB + alpha
+/// RGB32F  - 32-bit floating RGB
+/// RGBA32F - 32-bit floating RGB + alpha
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize, Hash)]
+#[cfg_attr(feature = "python", pyclass)]
```

Comment:
I'll double-check, but my understanding was that the GIL is only implicitly held if we were passing around/storing the `ImageMode` in a `PyAny`, which is restricted to a GIL lifetime, and since we're converting to a plain Rust `ImageMode` at the `PyDataType` boundary and only storing that pure Rust `ImageMode`, that definitely shouldn't be implicitly holding onto the GIL.

Am I misunderstanding something there?